go/cmd/ocitool/createlayer_cmd.go

```diff
@@ -70,10 +82,14 @@ func CreateLayerCmd(c *cli.Context) error {
 	// Need to flush before we count bytes and digest, might as well close since
 	// it's not needed anymore.
 	tw.Close()
-	gw.Close()
+	if config.UseZstd {
```

Comment:
Suggestion: Consider making the configuration generic WRT compression -- that is, you should check for something like:

```go
if config.compression = COMPRESSION_ZSTD {
```

instead.  May be overkill, though.

Reply:
I like this approach (and the other comments that follow suite).  I think this sets up the code infrastructure to scale better in the future for additional compression tools to be supported.